pyboreas/utils/odometry.py

```diff
@@ -386,16 +387,49 @@ def get_path_from_Tvi_list(T_vi_odom, T_vi_gt):
         path_odom (np.ndarray): K x 3 numpy array of estimated xyz coordinates
         path_gt (np.ndarray): K x 3 numpy array of groundtruth xyz coordinates
     """
-
     assert len(T_vi_odom) == len(T_vi_gt)  # assume 1:1 correspondence
     T_iv_odom = [np.linalg.inv(T_vk_i_odom) for T_vk_i_odom in T_vi_odom]
-
     T_iv_gt = [np.linalg.inv(T_vk_i_gt) for T_vk_i_gt in T_vi_gt]
-    T_odom_gt_i = T_iv_odom[0] @ np.linalg.inv(T_iv_gt[0])  # align the first pose
```

Comment:
What is this change trying to accomplish?

Reply:
Sorry for not cleaning this up properly. The reason for this change was to provide the path resolved in the first odometry frame instead of the first groundtruth frame. This helps with plot clarity, as all trajectories now start at (0, 0) instead of at the first gps position. This is especially nice when you run different sequences with very different starting gps positions.

If you are okay with that I will tidy up the code.

Reply:
Okay so I wasn't fully correct previously about why this change works, and I have a new proposed change.

I've attached the math behind the old and current (in this branch) get_path_from_Tvi_list() function written out in terms of the frames. I also provided code line numbers for reference.

The old format aligned everything to the **estimate inertial frame**. This can be different depending on the estimation pipeline, which is where I was having issues. Specifically, I was using a different estimator and wanted to plot the results on top of the results generated using vtr3 using get_path_from_Tvi_list(). My estimator inertial frame was different, and so the two extracted paths didn't align.

The current change aligns everything to the **groundtruth initial vehicle frame**. This is constant for a given sensor, and so I was able to overlay my different odometry estimates. However, this is naturally not constant for different sensors. So if you wanted to plot lidar and radar results generated using the same estimation pipeline and using the same inertial frame (as we do in vtr3) on the same plot, they would cease to align. 

The proposed solution I have is to thus resolve the odometry estimates relative to the **groundtruth inertial frame**, as shown in the final "Proposed" screenshot. This fixes both shortcomings of the previous 2 approaches, since the inertial frame of our groundtruth is always the same, regardless of estimation approach and regardless of sensor. The only issue is that the x/y coordinates are then relative to the groundtruth inertial frame, which for far-away sequences is some large number and  looks messy in the plot. However, we can simply 0 out the x/y coordinates for plotting purposes. I've verified this graphically for plotting vtr3 radar output, vtr3 lidar output, and my own pipeline radar odometry output.

If this makes sense to you, I'll make the changes for the new PR.

![old_current](https://github.com/user-attachments/assets/8f9c3e93-b975-4aa4-98e0-480ff9da3371)
![proposed](https://github.com/user-attachments/assets/98eb7510-84be-4dd1-a3a5-bae0d322a54b)

Reply:
Sure, looks good to me. And yeah, subtract the translation at the start from both if it's not (0,0,0).